src/libexpr/primops.cc

```diff
@@ -878,6 +878,33 @@ static RegisterPrimOp primop_trace({
 });
 
 
+/* Evaluate the first expression and print it on standard error if talkative verbosity
+ * is enabled. Then return the second expression.  Useful for debugging. */
+static void prim_debug(EvalState & state, const Pos & pos, Value * * args, Value & v)
+{
+    state.forceValue(*args[0], pos);
+    if (verbosity >= lvlTalkative) {
+        if (args[0]->type() == nString)
+            printError("debug: %1%", args[0]->string.s);
+        else
+            printError("debug: %1%", *args[0]);
+    }
+    state.forceValue(*args[1], pos);
+    v = *args[1];
+}
+
+static RegisterPrimOp primop_debug({
+    .name = "__debug",
```

Comment:
How is this different from `builtins.trace`?

Reply:
It works only if `-vv` is passed. Basically, Nix lang level debug logging.

Reply:
I see.